### PR TITLE
Payara 7 and Jakarta EE 11 support

### PR DESCRIPTION
**Pull request by jGauravGupta**
This PR introduces the following changes to support Payara 7 and Jakarta EE 11 in Apache NetBeans:

**Updated Build Type Handling:**

Changed the method signature from `short getBuild()` to `String getBuild()`. This change allows the system to recognize and list the latest `Alpha` versions of Payara 7: https://mvnrepository.com/artifact/fish.payara.distributions/payara/7.2024.1.Alpha1

**Alpha/Beta Version Filtering:**

Only the latest `Alpha` or `Beta` version will now be listed in the available versions. Other `Alpha` or `Beta` versions will be skipped to maintain a cleaner and more relevant version list.
image

**Archetype Update for Payara Micro:**

The archetype `fish.payara.starter:payara-starter-archetype` will now be used for the Payara Micro 6+.

### CODE REVIEW

File: enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/PayaraPlatformVersion.java [hunk 1]
```diff
@@ -387,7 +392,7 @@ public boolean equals(final PayaraPlatformVersionAPI version) {
             return this.major == version.getMajor()
                     && this.minor == version.getMinor()
                     && this.update == version.getUpdate()
-                    && this.build == version.getBuild();
+                    && (this.build == null ? version.getBuild() == null : this.build.equals(version.getBuild()));
```
Comment (matthiasblaesing) [inline enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/PayaraPlatformVersion.java:395]:
```suggestion
                    && Objects.equals(this.build, version.getBuild()));
```

File: enterprise/payara.tooling/test/unit/src/org/netbeans/modules/payara/tooling/data/PayaraVersionTest.java [hunk 2]
```diff
@@ -86,16 +86,17 @@ public void testToFullString() {
...
                 major  = Short.parseShort(numbers[0]);
                 minor  = Short.parseShort(numbers[1]);
                 update = Short.parseShort(numbers[2]);
-                build  = Short.parseShort(numbers[3]);
+                build  = numbers[3];
                 assertTrue(major == version.getMajor()
                         && minor == version.getMinor()
                         && update == version.getUpdate()
-                        && build == version.getBuild());
+                        && (build == null ? version.getBuild() == null : build.equals(version.getBuild())));
```
Comment (matthiasblaesing) [inline enterprise/payara.tooling/test/unit/src/org/netbeans/modules/payara/tooling/data/PayaraVersionTest.java:99]:
```suggestion
                        && Objects.equals(build, version.getBuild()));
```

**Review by matthiasblaesing — COMMENTED** [review]
I only eyeballed this, but looks sane to me. I left a few inline comments, but nothing major.

File: enterprise/payara.micro/src/org/netbeans/modules/fish/payara/micro/project/resources/action-mapping.xml [hunk 3]
```diff
@@ -30,7 +30,7 @@
             <goal>compiler:compile</goal>
             <goal>war:exploded</goal>
             <goal>payara-micro:stop</goal>
-            <goal>payara-micro:start</goal>
+            <goal>payara-micro:dev</goal>
```
[the remark this answers is not in the document]
Comment (jGauravGupta) [reply enterprise/payara.micro/src/org/netbeans/modules/fish/payara/micro/project/resources/action-mapping.xml]: `payara-micro:dev` and `payara-micro:start -DdeployWar=true -Dexploded=true -DautoDeploy=true -DliveReload=true -DkeepState=true -DtrimLog=true` perform the same operation. Updated the PR and used the latter, as it ensures backward compatibility with older versions, where `autoDeploy` and other development mode properties will be silently ignored.

**Review by matthiasblaesing — APPROVED** [review]
@jGauravGupta thank you for the updates. If this works for you, I'm fine with it. I suggest to squash and get this merged.

**Comment by jGauravGupta** [comment]
image
Squashed locally as squash is disabled in the Github.

**Comment by matthiasblaesing** [comment]
@jGauravGupta yes squashing on github was explicitly disabled as github has the tendency to destroy the author information. In contrast to the git CLI client github tries to deduce the author information from the public info of the profile of the author, which might be incomplete/hidden or just plain wrong.

**Pull request merged**